### ISSUE TITLE
Turn `_operation_id_no_change` into a fixed column for block machines

### ISSUE THREAD

**Issue by georgwiese** [powdr-labs/powdr#629]
`_operation_id_no_change` is a witness column generated by Powdr, constrained as:
```
_operation_id_no_change = ((1 - _block_enforcer_last_step) * (1 - <Latch>));
```

In the case of block machines, the latch is a fixed column. Because this polynomial only depends on fixed polynomials, it can itself be fixed. This reduces the cost of a block machine (the column doesn't need to be committed to at proving time) and also fixes problems in witness generation, see #626.

**Comment by Schaeff** [comment]
Could this be done by the optimiser?

**Comment by chriseth** [comment]
Yes, we should do that in the optimizer! Or maybe a different component that just gets a bunch of columns and tells us which is which.
